homework6.rb

```diff
@@ -0,0 +1,187 @@
+print arr.uniq
+
+
+# problem 7
+#Tell what's the major difference between an Array and a Hash?
+#
+# Arrays are ordered
+# Array index starts at 0
+#
+# Hash are unordered
+# HAsh is a collection  of key-value pairs
+# Hash index can be any object type
+# Hash enumerate their values in the order that the corresponding  keys were inserted
+puts
+
+# problem 8
+#Suppose you have a hash h = { a:1, b:2, c:3, d:4 }
+h = { a:1, b:2, c:3, d:4 }
+puts
+
+
+# problem 9
+#Can hash values be arrays? Yes  hash = {}; hash = { "1":[1,2,3], "2":[ "a", "b", "c"], "c":["ruby", 11, 22, "rails"]}
+# Can you have an array of hashes? (give examples) Yes
+#
+h= {"firstname" => "Nicolae", "lastname" => "SPINU", "age" => "20", "city"=>"CHISINAU"}
+h2= {"firstname" => "OLESEA", "lastname" => "SPINU", "age" => "18", "city"=>"MOSKOW"}
+array =[]
+array << h
+array << h2
+print array
+puts
+
+
+# problem 10
+#Given the following data structures.
+# Write a program that copies the information from the array into the empty hash that applies to the correct person.
+#
+contact_data = [["joe@email.com", "123 Main st.", "555-123-4567"], ["sally@email.com", "404 Not Found Dr.", "123-234-3454"]]
+contacts = {"Joe Smith" => {}, "Sally Johnson" => {}}
+
+element =contact_data[0]
+contacts["Joe Smith"]={"email"=> element[0], "address"=> element[1], "phone"=> element[2]}
+ element1 =contact_data[1]
+contacts["Sally Johnson"]={"email"=> element1[0], "address"=> element1[1], "phone"=> element1[2]}
+puts contacts
+puts
+
+
+# problem 11
+# Using the hash you created from the previous exercise,
+# demonstrate how you would access Joe's email and Sally's phone number?
+puts contacts["Joe Smith"]["email"]
+puts contacts["Sally Johnson"]["phone"]
+puts
+
+# problem 12
+# Use Ruby's Array method delete_if and String method start_with?
+# to delete all of the words that begin with an "s" in the following array.
+#arr = ['snow', 'winter', 'ice', 'slippery', 'salted roads', 'white trees']
+arr = ['snow', 'winter', 'ice', 'slippery', 'salted roads', 'white trees']
+arr.delete_if { |word| word.start_with?("s") }
+print arr
+puts
+
+#Then recreate the arr and get rid of all of the words that start with "s" or starts with "w".
+arr = ['snow', 'winter', 'ice', 'slippery', 'salted roads', 'white trees']
+arr.delete_if { |word|word.start_with?("s","w") }
+print arr
+puts
+
+# problem 13
+#Take the following array:
+#a = ['white snow', 'winter wonderland', 'melting ice','slippery sidewalk', 'salted roads', 'white trees']
+#and turn it into a new array that consists of strings containing one word.
+# (ex. ["white snow", etc...] → ["white", "snow", etc...].
+# Look into using Array's map and flatten methods, as well as String's split method.
+a = ['white snow', 'winter wonderland', 'melting ice','slippery sidewalk', 'salted roads', 'white trees']
+arr = []
+ #arr+= a.each { |el| el.split(" ") }   PLEASE Explain Why this not working????
+a.map do |word|
```

Comment:
You could use `each` for this iteration since you're iterating through `a` array without changing it